lib/index.js

```diff
@@ -86,22 +86,15 @@ internals.authenticate = async function(token, options, request, h) {
   let tokenType = options.tokenType || 'Token'; // see: https://git.io/vXje9
   let decoded;
 
-  if (!token && request.auth.mode === 'optional') {
-    return {
-      error: internals.raiseError(options, 'unauthorized', null, tokenType),
-      payload: {
-        credentials: tokenType,
-      },
-    };
-  }
-
   if (!token) {
     return {
       error: internals.raiseError(
         options,
         'unauthorized',
         'token is null',
-        tokenType
+        tokenType,
+        null,
+        true
```

Comment:
function definition for `internals.raiseError` accepts five params:
https://github.com/dwyl/hapi-auth-jwt2/blob/ed89eaba60bf167b3786d9b419927eb3b44e6e04/lib/index.js#L245-L251
we appear to be passing through 6 arguments. 💭

Reply:
Please check the new signature of the raiseError function, the `isMissingToken` parameter was added so that we could signal to the parent framework directly that we found no valid token and additional schemes (if present) should be checked rather than relying on raising errors with no message. This does however cause our plugin to report failed auth in the console when others succeed. I'd happily defer to your preference on this, either we silently fail a missing token for single scheme auth or over zealously report auth failures in multi scheme auth. What do you think?

Reply:
![image](https://user-images.githubusercontent.com/194400/67083743-9c859f00-f193-11e9-9f56-9e7871a8226f.png)

👍